SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitApiUnitTest.java

```diff
@@ -179,4 +181,88 @@ public Void answer(InvocationOnMock invocation) {
 
         verify(createMandateCallback).onResult((CreateMandateResult) isNull(), same(exception));
     }
+
+    @Test
+    public void tokenize_buildsHttpRequest_withAllParams() throws JSONException {
+        SEPADirectDebitApi sut = new SEPADirectDebitApi(httpClient);
+        sut.tokenize("1234", "a-customer-id", "a-bank-reference-token", "ONE_OFF", sepaDirectDebitTokenizeCallback);
+
+        ArgumentCaptor<HttpRequest> captor = ArgumentCaptor.forClass(HttpRequest.class);
+        verify(httpClient).sendRequest(captor.capture(), ArgumentMatchers.<HttpResponseCallback>any());
+
+        HttpRequest httpRequest = captor.getValue();
+        assertEquals("POST", httpRequest.getMethod());
+        assertEquals("application/json", httpRequest.getHeaders().get("Content-Type"));
+        assertEquals("development_testing_pwpp_multi_account_merchant", httpRequest.getHeaders().get("Client-Key"));
+        assertEquals("merchants/pwpp_multi_account_merchant/client_api/v1/payment_methods/sepa_debit_accounts", httpRequest.getPath());
+
+        JSONObject data =new JSONObject(new String(httpRequest.getData()));
+        JSONObject accountData = data.getJSONObject("sepa_debit_account");
+        assertEquals("a-customer-id", accountData.get("customer_id"));
+        assertEquals("1234", accountData.get("iban_last_chars"));
+        assertEquals("ONE_OFF", accountData.get("mandate_type"));
+        assertEquals("a-bank-reference-token", accountData.get("bank_reference_token"));
+    }
+
+    @Test
+    public void tokenize_onSuccessfulHttpResponse_callsBackSEPADirectDebitNonce() {
+        doAnswer(new Answer<Void>() {
```

Comment:
We will clean up these tests with a MockBuilder once the feature is in Sandbox and we aren't directly integrating with `HttpClient`